File: 07_Python/rock_paper_scissors_game.py
import random

def play_game():

    user_pts = 0
    com_pts = 0
    options = ["R", "P", "S"]

    name = input("What's your name?: ")
    print(f"Hello! {name}. Let's play rock-paper-scissors.")
    print("""Instruction: Type 'R' for rock, 'P' for paper, and 'S' for scissors.
If you want to end the game, type 'exit'.""")
    
    while True:
        user_select = input("Choose one: ")
        user_select = user_select.upper()
        com_select = random.choice(options)

        if user_select == com_select:
            user_pts += 0
            com_pts += 0
            print(f"{user_select} vs {com_select}... Tie!")
            print(f"Your score: {user_pts}  Computer score: {com_pts}")

        elif user_select == "R" and com_select == "P":
            user_pts += 0
            com_pts += 1
            print(f"{user_select} vs {com_select}... You lost.")
            print(f"Your score: {user_pts}  Computer score: {com_pts}")

        elif user_select == "R" and com_select == "S":
            user_pts += 1
            com_pts += 0
            print(f"{user_select} vs {com_select}... You won!")
            print(f"Your score: {user_pts}  Computer score: {com_pts}")

        elif user_select == "P" and com_select == "S":
            user_pts += 0
            com_pts += 1
            print(f"{user_select} vs {com_select}... You lost.")
            print(f"Your score: {user_pts}  Computer score: {com_pts}")

        elif user_select == "P" and com_select == "R":
            user_pts += 1
            com_pts += 0
            print(f"{user_select} vs {com_select}... You won!")
            print(f"Your score: {user_pts}  Computer score: {com_pts}")

        elif user_select == "S" and com_select == "R":
            user_pts += 0
            com_pts += 1
            print(f"{user_select} vs {com_select}... You lost.")
            print(f"Your score: {user_pts}  Computer score: {com_pts}")

        elif user_select == "S" and com_select == "P":
            user_pts += 1
            com_pts += 0
            print(f"{user_select} vs {com_select}... You won!")
            print(f"Your score: {user_pts}  Computer score: {com_pts}")

        elif user_select == "EXIT":
            if user_pts == com_pts:
                return(f"Tie! Good game. Your score: {user_pts}  Computer score: {com_pts}")
            elif user_pts > com_pts:
                return(f"Congrats! You won. Your score: {user_pts}  Computer score: {com_pts}")
            else:
                return(f"You lost. Let's play again next time. Your score: {user_pts}  Computer score: {com_pts}")
        
        else:
            print("Invalid input. Please try again.")

play_game()

File: 02_R/02.1_Programming/rock_paper_scissors_game.R
play_game <- function() {
  
  print("Instruction: type 'exit' to exit the program")
  
  options <- c("rock", "paper", "scissors")
  print(options)
  
  user_score <- 0
  computer_score <- 0
  
  while (TRUE) {
    
    user_select <- readline("Choose one: ")
    computer_select <- sample(options, 1)
    
    if (user_select == computer_select ) {
      print(computer_select)
      print("Tie!")
      user_score = user_score + 0
      computer_score = computer_score + 0
    } else if ((user_select == 'rock' & computer_select == 'paper') |
               (user_select == 'paper' & computer_select == 'scissors') |
               (user_select == 'scissors' & computer_select == 'rock')) {
      print(computer_select)
      print("You lost. Don't give up!")
      user_score = user_score + 0
      computer_score = computer_score + 1
    } else if ((user_select == 'rock' & computer_select == 'scissors')|
               (user_select == 'paper' & computer_select == 'rock') |
               (user_select == 'scissors' & computer_select == 'paper')) {
      print(computer_select)
      print("You won. Way to go!")
      user_score = user_score + 1
      computer_score = computer_score + 0
    } else if (user_select == 'exit') {
      print("Good game! Thanks for playing.")
      
        if (user_score == computer_score) {
          print(paste("Tie. You scored ", user_score, " and I scored ", computer_score))
        } else if (user_score < computer_score) {
          print(paste("You lost. You scored ", user_score, " and I scored ", computer_score))
        } else {
          print (paste("Congrats, you won!. You scored ", user_score, " and I scored ", computer_score))
        }
        break
    } else {
      print("Invalid iput. Please try again.")
    }
  } 
}
